arangosh/Shell/V8ClientConnection.cpp

```diff
@@ -1735,52 +1735,98 @@ v8::Local<v8::Value> V8ClientConnection::patchData(
   return requestData(isolate, fu::RestVerb::Patch, location, body, headerFields);
 }
 
-v8::Local<v8::Value> V8ClientConnection::requestData(
-    v8::Isolate* isolate, fu::RestVerb method, arangodb::velocypack::StringRef const& location,
-    v8::Local<v8::Value> const& body,
-    std::unordered_map<std::string, std::string> const& headerFields, bool isFile) {
+int fuerteToArangoErrorCode(fu::Error ec) {
+  ErrorCode errorNumber = TRI_ERROR_NO_ERROR;
+  switch (ec) {
+    case fu::Error::CouldNotConnect:
+    case fu::Error::ConnectionClosed:
+      errorNumber = TRI_ERROR_SIMPLE_CLIENT_COULD_NOT_CONNECT;
+      break;
+
+    case fu::Error::ReadError:
+      errorNumber = TRI_ERROR_SIMPLE_CLIENT_COULD_NOT_READ;
+      break;
+
+    case fu::Error::WriteError:
+      errorNumber = TRI_ERROR_SIMPLE_CLIENT_COULD_NOT_WRITE;
+      break;
 
-  bool retry = true;
-
-again:
-  auto req = std::make_unique<fu::Request>();
-  req->header.restVerb = method;
-  req->header.database = _databaseName;
-  req->header.parseArangoPath(location.toString());
+    default:
+      errorNumber = TRI_ERROR_SIMPLE_CLIENT_UNKNOWN_ERROR;
+      break;
+  }
+  return static_cast<int>(errorNumber);
+}
+
+// V8 -> fuerte
+void translateHeaders(fu::Request& request,
+                      fu::RestVerb const method,
+                      arangodb::velocypack::StringRef const& location,
+                      std::string const& databaseName,
+                      bool forceJson,
+                      std::chrono::duration<double> const& requestTimeout,
+                      std::unordered_map<std::string, std::string> const& headerFields) {
+  request.header.restVerb = method;
+  request.header.database = databaseName;
+  request.header.parseArangoPath(location.toString());
   for (auto& pair : headerFields) {
     if (boost::iequals(StaticStrings::ContentTypeHeader, pair.first)) {
       if (pair.second == StaticStrings::MimeTypeVPack) {
-        req->header.contentType(fu::ContentType::VPack);
+        request.header.contentType(fu::ContentType::VPack);
       } else if ((pair.second.length() >= StaticStrings::MimeTypeJsonNoEncoding.length()) &&
                (memcmp(pair.second.c_str(),
                        StaticStrings::MimeTypeJsonNoEncoding.c_str(),
                        StaticStrings::MimeTypeJsonNoEncoding.length()) == 0)) {
         // ignore encoding etc.
-        req->header.contentType(fu::ContentType::Json);
+        request.header.contentType(fu::ContentType::Json);
       } else {
-        req->header.contentType(fu::ContentType::Custom);
+        request.header.contentType(fu::ContentType::Custom);
       }
     } else if (boost::iequals(StaticStrings::Accept, pair.first)) {
-      req->header.acceptType(fu::ContentType::Custom);
+      request.header.acceptType(fu::ContentType::Custom);
     }
```

Comment:
I am still not convinced that we need all this here, given that fuerte's header.addMeta(...) should do it already.

Reply:
its unneccessary, completely removed.
flipped sequence for `forceJson` so it is now an overrideable default.